bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/ChallengeResponse/CreateChallengeResponse.cs

```diff
@@ -0,0 +1,168 @@
+using ElectionGuard.Decryption.Challenge;
+using ElectionGuard.Decryption.Shares;
+using ElectionGuard.ElectionSetup;
+
+namespace ElectionGuard.Decryption.ChallengeResponse;
+
+
+// TODO: all of the iterator functions for al;l of these types should be consoldaited somewhere
+// in fact we can probably define a single base class collection lookup
+// that takes a generic and iterates over the collectiosn and does zips and comaprisons
+
+/// <summary>
+/// Decryption extension methods for the <see cref="Guardian" /> class
+/// </summary>
+public static class GuardianChallengeResponseExtensions
+{
+
+    /// <summary>
+    /// Compute a challenge response for a specific guardian challenge that was issued by an election administrator.
+    /// </summary>
+    public static GuardianChallengeResponse ComputeChallengeResponse(
+        this Guardian guardian,
+        GuardianShare share, GuardianChallenge challenge)
+    {
+        if (guardian.CommitmentOffset is null)
```

Comment:
this CommitmentOffset is a bit awkward. Basically, using a similar process to verifying the guardian key shares on the polynomial, we need to calculate this value to be used ina few places. Equation 10 in the v1.53 spec shows a bunch of equality comparisons for computing `g^Pi(x)` and this is one of those processes. It also has to be done for ALL guardians, even if they are not participating in the decryption so it made the most sense to add it to the guardian object.

its possible that there is an issue with accessing this value because I'm not 100% up to speed on how the rehydration works for the guardian objects - so i tossed it into the Guardian.cs class asa computed value. hopefully that works well.